### ISSUE TITLE
Collections' `avg()` do not work as expected

### ISSUE THREAD

**Issue by aarreedd** [laravel/internals#337]
`avg()` does not work as expected if the collection contains a null value. Right now the null is treated as a zero, but I think it should be skipped all together. Is this the correct behavior?

Example:

    $collect = collect([100, null, 200,  300]);
    $avg = $collect->avg();
    echo $min; // 150, but I think it should be 200

**Comment by Garbee** [comment]
This is working as expected. `null` casts to `0`, but is still an item in the array so the count to divide by is *4* and not *3*. Just like how [PHP handles summing](https://3v4l.org/2SSn0). If you don't want to count null, then reject them before averaging. `$col->reject(function($item) { return $item === null; })->avg();`

**Comment by tomschlick** [comment]
Yeah I'd agree. This is a feature not a bug.

**Comment by aarreedd** [comment]
Yeah, that makes sense.

**Comment by aarreedd** [comment]
If we decided to `filter()` the `min()` and `max()` shouldn't we run `filter()` on `avg()` too?

**Comment by Garbee** [comment]
No, because `null` is 0 and it is an item in the index. It is a different context than trying to get the *min* value and not actually getting the min value.

Here what you expect is entirely dependent upon what you are trying to do. There is no clear point where things are not doing what they are supposed to. As, we are literally doing what PHP would do with the same values. If anything this should just be a documentation note, but nothing more.

**Comment by Garbee** [comment]
FTR, I explicitly asked Taylor on Twitter to look at the `min` issue since looking over it again just now I realized how critical of a bug it was. That could affect production apps in some pretty severe ways for businesses expecting it to do what it says it will do right away.

This however is a possible pitfall that developers just need to understand they'll need to guard against if they expect it to happen.
